pkg/trigram/trigram.go

```diff
@@ -9,19 +12,38 @@ type Trigram struct {
 	Offset uint64
 }
 
+func normalizeToAscii(s string) (string, map[int]int) {
+	ogOffsets := make(map[int]int)
+
+	var b strings.Builder
+	norm := norm.NFKD.String(s)
+
+	for i, r := range norm {
+		if r <= unicode.MaxASCII {
+			b.WriteRune(r)
+
+			fmt.Printf("valid: %v\n", i)
+		} else {
+			fmt.Printf("invalid: %v\n", i)
+		}
+	}
+	return b.String(), ogOffsets
+}
+
```

Comment:
I've been picking my hair for the past hour trying to come up with an elegant solution. 

`normalizeToAscii` should normalize the string while returning a mapping of the original offers to the new index. 


this is because given `"h∫owd*y)__do "`, it normalizes it to `"howdydo "`, which is great. but we lose the offsets.


The problem right now is that since we're iterating through the unicode, the `i` doesn't match up. a sample output looks like:
```
valid: 0
invalid: 1
valid: 4
valid: 5
valid: 6
invalid: 7
valid: 10
```


Since invalid characters take more than one element.